easybuild/framework/easyblock.py

```diff
@@ -514,9 +547,19 @@ def fetch_extension_sources(self, skip_checksums=False):
 
                     if ext_options.get('nosource', None):
                         exts_sources.append(ext_src)
+                    elif ext_options.get('sources', None):
+                        sources = ext_options['sources']
+                        if isinstance(sources, (list, tuple)):
+                            raise EasyBuildError("'sources' entry to 'exts_list' must be a single dictionary.")
```

Comment:
I would allow single-element lists, to minimize the gap with the `sources` easyconfig parameter, see https://github.com/kelseymh/easybuild-framework/pull/2